openapi-cli/src/test/resources/ballerina-to-openapi/constraint/floatMax.bal

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2023, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/constraint;
+
+@constraint:Float {
+    maxValueExclusive: 10.5,
+    minValue: 2.5
+}
+
+public type Rating float;
+
+public type Hotel record {
+    Rating rate;
+};
+
+service /payloadV on new http:Listener(9090) {
+    resource function post pet(@http:Payload Hotel body) returns error? {
```

Comment:
Currently, there are two ways to define constraint annotations:

1. On type definitions:

    ```bal
    @constraint:Float {
        maxValueExclusive: 10.5,
        minValue: 2.5
    }
    public type Rating float;
    ```

2. On record field type:

    ```bal
    public type Hotel record {
        @constraint:Float {
            maxValueExclusive: 10.5,
            minValue: 2.5
        }
        float rate;
    };
    ```

The tests here only address the first, we need tests for the second one as well

Also the payload can be directly a basic type with constraint annotations, so we need to address that as well
```bal
resource function post pets/[int id]/rating(Rating rating) returns error? 
```